hollow/src/main/java/com/netflix/hollow/core/write/objectmapper/DeprecatedApi.java

```diff
@@ -1,15 +1,13 @@
 package com.netflix.hollow.core.write.objectmapper;
 
 import static java.lang.annotation.ElementType.FIELD;
-import static java.lang.annotation.ElementType.METHOD;
 
 import java.lang.annotation.Retention;
 import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
 
-
 @Retention(RetentionPolicy.RUNTIME)
-@Target({FIELD, METHOD})
+@Target(FIELD)
 public @interface DeprecatedApi {
```

Comment:
We should document this annotation, otherwise developers will not know how/where to use it and what the side-effects are.  I wish we could somehow reuse the existing @Deprecated annotation with JavaDoc but we loose the latter when reflectively processing.  Should it be recommended this annotation be used in conjunction with @Deprecated?

I wonder this annotation should not be directly passed through to the `HollowObjectSchema.addField` methods, since that abstraction is generally free of Java's reflective types.  More generally perhaps there will be other kinds of Hollow schema field modifier in the future?

Reply:
> I recommend just adding one new method HollowObjectSchema.addField, given the use of deprecation is likely rare.

I need two of the `addField` method declarations in `HollowObjectTypeMapper`. The third one is the result of "telescoping".

> I wonder if we can generalize this to pass in a Set of some "meta-data" (Hollow field schema annotation), since deprecation is just one of many possible (not yet thought up) field modifiers or annotations. That relates to a comment on the @DeprecatedApi annotation.

Annotation processing is one of the scopes that can be took in consideration. If you would like to extend existing solution, which actually pass only types, there should be same abstract placeholder for passed through `meta-data`. 
 
> We should document this annotation, otherwise developers will not know how/where to use it and what the side-effects are.

Will do.

>  I wish we could somehow reuse the existing @deprecated annotation with JavaDoc but we loose the latter when reflectively processing. Should it be recommended this annotation be used in conjunction with @deprecated?

The goal of this PR is to provide possibility of using @Deprecated with JavaDoc comment, thus this specific annotation should be strictly related to the Java's @Deprecated.

> I wonder this annotation should not be directly passed through to the HollowObjectSchema.addField methods, since that abstraction is generally free of Java's reflective types. More generally perhaps there will be other kinds of Hollow schema field modifier in the future?

Previously mentioned abstract 'place-holder' will be enough? What future modifiers it should contains?

Reply:
> I need two of the addField method declarations in HollowObjectTypeMapper. The third one is the result of "telescoping".